source/numerics/data_out.cc

```diff
@@ -338,6 +338,47 @@ void DataOut<dim,DoFHandlerType>::build_patches
   Assert (n_subdivisions >= 1,
           Exceptions::DataOut::ExcInvalidNumberOfSubdivisions(n_subdivisions));
 
+#ifdef DEBUG
+  {
+    // Check that names for datasets are only used once. This is somewhat
+    // complicated, because vector ranges might have a name or not.
+    std::set<std::string> all_names;
```

Comment:
If you put everything that's here into a separate function in `DataOutDoFData`, then you can (i) just put the call to the function here into an `Assert(...)`, and (ii) use the same function in `DataOutFaces` and `DataOutRotation` (and `DataOutStack`, I suppose).